atlasdb-timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/PaxosTimeLockServerIntegrationTest.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.timelock;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.File;
+import java.util.SortedMap;
+
+import javax.net.ssl.SSLSocketFactory;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.AfterClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSortedMap;
+import com.palantir.atlasdb.http.AtlasDbHttpClients;
+import com.palantir.atlasdb.timelock.config.TimeLockServerConfiguration;
+import com.palantir.lock.LockDescriptor;
+import com.palantir.lock.LockMode;
+import com.palantir.lock.LockRefreshToken;
+import com.palantir.lock.LockRequest;
+import com.palantir.lock.RemoteLockService;
+import com.palantir.lock.StringLockDescriptor;
+import com.palantir.timestamp.TimestampService;
+
+import io.dropwizard.testing.ResourceHelpers;
+import io.dropwizard.testing.junit.DropwizardAppRule;
+
+public class PaxosTimeLockServerIntegrationTest {
+    private static final String NOT_FOUND_CODE = "404";
+
+    private static final String CLIENT_1 = "test";
+    private static final String CLIENT_2 = "test2";
+    private static final String NONEXISTENT_CLIENT = "nonexistent";
+    private static final String INVALID_CLIENT = "test2\b";
+
+    private static final Optional<SSLSocketFactory> NO_SSL = Optional.absent();
+    private static final String LOCK_CLIENT_NAME = "lock-client-name";
+    private static final SortedMap<LockDescriptor, LockMode> LOCK_MAP = ImmutableSortedMap.of(
+            StringLockDescriptor.of("lock1"), LockMode.WRITE);
+
+    @ClassRule
+    public static final DropwizardAppRule<TimeLockServerConfiguration> APP = new DropwizardAppRule<>(
+            TimeLockServer.class,
+            ResourceHelpers.resourceFilePath("paxosSingleServer.yml"));
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+        FileUtils.deleteDirectory(new File("var/test/data"));
```

Comment:
@hsaraogi might mean "on the build server"?

Also, the safe thing to do here is to use a `@TemporaryFolder`.